test/integration/xds_integration_test.cc

```diff
@@ -36,5 +40,46 @@ TEST_P(XdsIntegrationTest, RouterRequestAndResponseWithBodyNoBuffer) {
   testRouterRequestAndResponseWithBody(1024, 512, false);
 }
 
+typedef HttpProtocolIntegrationTest LdsIntegrationTest;
+
+INSTANTIATE_TEST_SUITE_P(Protocols, LdsIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams(
+                             {Http::CodecClient::Type::HTTP1}, {FakeHttpConnection::Type::HTTP1})),
+                         HttpProtocolIntegrationTest::protocolTestParamsToString);
+
+// Sample test making sure our config framework correctly reloads listeners.
+TEST_P(LdsIntegrationTest, ReloadConfig) {
+  autonomous_upstream_ = true;
+  initialize();
+  // Given we're using LDS in this test, initialize() will not complete until
+  // the initial LDS file has loaded.
+  EXPECT_EQ(1, test_server_->counter("listener_manager.lds.update_success")->value());
+
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+
+  // HTTP 1.0 is disabled by default.
+  std::string response;
+  sendRawHttpAndWaitForResponse(lookupPort("http"), "GET / HTTP/1.0\r\n\r\n", &response, true);
+  EXPECT_TRUE(response.find("HTTP/1.1 426 Upgrade Required\r\n") == 0);
+
+  // Create a new config with HTTP/1.0 proxying.
+  ConfigHelper new_config_helper(version_, *api_,
+                                 MessageUtil::getJsonStringFromMessage(config_helper_.bootstrap()));
+  new_config_helper.addConfigModifier(
+      [&](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager& hcm) {
+        hcm.mutable_http_protocol_options()->set_accept_http_10(true);
+        hcm.mutable_http_protocol_options()->set_default_host_for_http_10("default.com");
+      });
+
+  // Create an LDS response with the new config, and reload config.
+  new_config_helper.setLds("1");
```

Comment:
Can we just have a `new_config_helper.reloadLds()` that both bumps version and waits for update? This might need to have finger grained methods like you have in some times, but most just want to "make it so".

Reply:
I don't think that's the way to go about it.  The config helper is all about config - it knows nothing about the running Envoy, how to get stats etc.  
We could have a test framework wrapper which does this, but given it is 2 lines of code I'm inclined to add it when/if we need it.

Reply:
Yeah, not in the config helper itself, instead some stateful helper that hides this and owns the config helper. I think it's not about solving a problem, but about making tests more declarative and readable. Less line noise ==> more good :)

Reply:
Yeah, again with the idea of multiple config changes in mind I think that will be a step back in readability.  It'd help to have more use cases so again I'm inclined to see how this is used, and wrapper commonly reused code rather than implement something I have a hunch will make it harder to use.  Again, open to doing it if that's what it takes to land this.

Reply:
I'll defer to you on this PR. I still think that for the common case reducing boilerplate is a good thing, but let's merge this for now and iterate.